### ISSUE TITLE
Parameters tuning for quantile forest

### ISSUE THREAD

**Issue by zpunit331111111** [grf-labs/grf#811]
Hi, I'd like to tune the parameter 'mtry' in quantile_forest(). I am wondering if the parameter tuning feature is available because I did not see the 'tune.parameters' variable in the quantile forest function? thanks!

**Comment by erikcs** [comment]
Parameter tuning for quantile_forest is currently not implemented.

**Comment by zpunit331111111** [comment]
Then is there any way to find parameter values from the quantile_forest() models, such as the value of mtry? When I saved the data, I only save the rf = quantile_forest(x,y,....), but could not track back to the specific arguments used to build rf. thanks

**Comment by erikcs** [comment]
Sorry, that is currently not saved. Did you fit rf with default values for mtry? In that case the value will be `min(ceiling(sqrt(ncol(X)) + 20), ncol(X))`

**Comment by zpunit331111111** [comment]
yes, I did. So my data has 17 independent variables. I am wondering since it is smaller than 20, by using the default value of mtry, I actually used all variables for node split? Does this remove the randomness and make the model no different with ensembled trees? thanks!

**Comment by erikcs** [comment]
Hi @zpunit331111111, actually, `mtry` in GRF differs slightly from an "ordinary" random forest, from the [online reference](https://grf-labs.github.io/grf/REFERENCE.html#mtry):

>To more closely match the theory in the GRF paper, the number of variables considered is actually drawn from a poisson distribution with mean equal to mtry. A new number is sampled from the distribution before every tree split.

So in your case each 17 variables are not always considered at every split.

**Comment by halflearned** [comment]
Hi @zpunit331111111 , that's in interesting question!

I played around with a couple of simulation designs with rare categories, and your observation seems to be right: it's often the case that the forest ends up simply not splitting on these variables. If you won't be constructing confidence intervals, then you could increase subsample size by changing `sample.fraction` to something higher than 1 (you'll also have to set `ci.group.size` to 1), and/or disabling honesty by setting `honesty = FALSE`. 

However, you should keep in mind that variable splits are an imperfect diagnostic tool, and maybe there isn't much to worry about here. In particular, if your "categorical" variables are highly correlated with some other continuous variables, the forest may end up splitting a lot on the continuous variables, even though the resulting subgroups are similar to what you would have found had it split on the categorical ones. 

Finally, you may want to check out our sister package [`sufrep`](https://github.com/grf-labs/sufrep) and see if the methods there are of any help in improving your performance. However, at least in my simulations, I didn't notice a substantial change in prediction accuracy in a setting that is similar to what you described.

**Comment by zpunit331111111** [comment]
Hi @halflearned, thanks for your comments! I never used sample fraction larger than 1 before, and wondering if that’s something will in general impact prediction accuracy also, or just changing the variables importance ranking? Thank you so much!

**Comment by halflearned** [comment]
Hi @zpunit331111111. Right -- `sample.fraction` is a tuning parameter. If you are after prediction accuracy, you can try changing this parameter and checking whether that yields any improvement, just as you would with any other parameter. 

(Also, sorry: I meant you could try setting `sample.fraction` to something larger than 0.5, and not "1" as I originally wrote. I edited the comment above accordingly.)

**Comment by zpunit331111111** [comment]
Thanks @halflearned for clear that up! I got it now, really appreciate your help.

**Comment by zpunit331111111** [comment]
Hi @halflearned, I redo the models, and have some follow up questions. I compared the results with two model settings: 1)regression.splitting = TRUE and honesty = FALSE, 2) regression.splitting = FALSE , honesty = TRUE, clusters = individual_ID. All other parameters were the same. Since the GRF algorithms have the splitting rules that are more suitable to quantiles estimation, I was expecting the second setting has better performance (measured by both oob and new data using MSE, MAPE). However, the results said the opposite, although the difference is not that large. I am trying to understand the results, hence draw some conclusions of methods suitability for my specific problem. I suspected this maybe due to honesty since it reduced the data size for training, the data towards distribution tails might not be even selected (and for regression.split. I also read the content on the algorithm reference website that ‘it’s important to note that honesty may hurt performance when working with very small datasets...’ my data includes 300 repeated measurements for each individual ID, and there are about 700+ IDs. So for my case is this considered a small dataset?
